### ISSUE TITLE
socket.error, No such device

### ISSUE THREAD

**Issue by rlue** [enen92/script.tubecast#5]
Haven't investigated the issue very closely, but TubeCast just stopped working on my RPi 3B+ (w/ LibreElec) the other day. Getting the following error in the logs:

    23:12:21.974 T:1752081312   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                 - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                Error Type: <class 'socket.error'>
                                                Error Contents: (19, 'No such device')
                                                Traceback (most recent call last):
                                                  File "/storage/.kodi/addons/script.tubecast/main.py", line 7, in <module>
                                                    service.run()
                                                  File "/storage/.kodi/addons/script.tubecast/resources/lib/service.py"

---

**EDIT:** FWIW, the problem went away after a restart. Also FWIW, I'm on 1.0.6.

**Comment by enen92** [comment]
Can you provide a full debug log with the debug settings of the addon enabled as well? It is a bit hard to traceback the error

**Comment by rlue** [comment]
hm, (un?)fortunately, I haven't been able to reproduce the problem since it happened.

In the meantime, here's the excerpt from above with 50 lines of context above and below:

```
...
23:12:17.403 T:1945243648  NOTICE: Found resolution 1280 x 800 for display 0 with 1280 x 800 @ 60.000000 Hz
23:12:17.403 T:1945243648  NOTICE: Found resolution 1280 x 800 for display 0 with 1280 x 800 @ 59.940063 Hz
23:12:17.403 T:1945243648  NOTICE: Found resolution 1280 x 960 for display 0 with 1280 x 960 @ 60.000000 Hz
23:12:17.403 T:1945243648  NOTICE: Found resolution 1280 x 960 for display 0 with 1280 x 960 @ 59.940063 Hz
23:12:17.403 T:1945243648  NOTICE: Found resolution 1280 x 1024 for display 0 with 1280 x 1024 @ 60.000000 Hz
23:12:17.403 T:1945243648  NOTICE: Found resolution 1280 x 1024 for display 0 with 1280 x 1024 @ 59.940063 Hz
23:12:17.404 T:1945243648  NOTICE: Found resolution 1360 x 768 for display 0 with 1360 x 768 @ 60.000000 Hz
23:12:17.404 T:1945243648  NOTICE: Found resolution 1360 x 768 for display 0 with 1360 x 768 @ 59.940063 Hz
23:12:17.404 T:1945243648  NOTICE: Found resolution 1680 x 1050 for display 0 with 1680 x 1050 @ 60.000000 Hz
23:12:17.404 T:1945243648  NOTICE: Found resolution 1680 x 1050 for display 0 with 1680 x 1050 @ 59.940063 Hz
23:12:17.404 T:1945243648  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080 @ 60.000000 Hz
23:12:17.404 T:1945243648  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080 @ 59.940063 Hz
23:12:17.404 T:1945243648  NOTICE: Found resolution 1280 x 720 for display 0 with 1280 x 720 @ 60.000000 Hz
23:12:17.404 T:1945243648  NOTICE: Found resolution 1280 x 720 for display 0 with 1280 x 720 @ 59.940063 Hz
23:12:17.404 T:1945243648  NOTICE: Found resolution 720 x 480 for display 0 with 720 x 480 @ 60.000000 Hz
23:12:17.404 T:1945243648  NOTICE: Found resolution 720 x 480 for display 0 with 720 x 480 @ 59.940063 Hz
23:12:17.404 T:1945243648  NOTICE: Found resolution 720 x 576 for display 0 with 720 x 576 @ 50.000000 Hz
23:12:17.404 T:1945243648  NOTICE: Found resolution 1280 x 720 for display 0 with 1280 x 720 @ 50.000000 Hz
23:12:17.404 T:1945243648  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080 @ 50.000000 Hz
23:12:17.404 T:1945243648  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080 @ 24.000000 Hz
23:12:17.404 T:1945243648  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080 @ 23.976025 Hz
23:12:17.404 T:1945243648  NOTICE: Found (1920x1080@60.000000) at 33, setting to RES_DESKTOP at 16
23:12:17.404 T:1945243648  NOTICE: Checking resolution 16
23:12:17.581 T:1945243648  NOTICE: GL_VENDOR = Broadcom
23:12:17.581 T:1945243648  NOTICE: GL_RENDERER = VideoCore IV HW
23:12:17.581 T:1945243648  NOTICE: GL_VERSION = OpenGL ES 2.0
23:12:17.581 T:1945243648  NOTICE: GL_SHADING_LANGUAGE_VERSION = OpenGL ES GLSL ES 1.00
23:12:17.581 T:1945243648  NOTICE: GL_EXTENSIONS = GL_OES_compressed_ETC1_RGB8_texture GL_OES_compressed_paletted_texture GL_OES_texture_npot GL_OES_depth24 GL_OES_vertex_half_float GL_OES_EGL_image GL_OES_EGL_image_external GL_EXT_discard_framebuffer GL_OES_rgb8_rgba8 GL_OES_depth32 GL_OES_mapbuffer GL_EXT_texture_format_BGRA8888 GL_APPLE_rgb_422 GL_EXT_debug_marker
23:12:18.415 T:1858073504  NOTICE: Running database version Addons27
23:12:18.418 T:1858073504  NOTICE: Running database version ViewModes6
23:12:18.462 T:1858073504  NOTICE: Running database version Textures13
23:12:18.528 T:1816130464  NOTICE: Register - new cec device registered on cec->RPI: CEC Adapter (2708:1001)
23:12:18.531 T:1858073504  NOTICE: Running database version MyMusic60
23:12:18.582 T:1858073504  NOTICE: Running database version MyVideos107
23:12:18.586 T:1858073504  NOTICE: Running database version TV29
23:12:18.589 T:1858073504  NOTICE: Running database version Epg11
23:12:18.592 T:1945243648  NOTICE: start dvd mediatype detection
23:12:19.298 T:1945243648 WARNING: JSONRPC: Could not parse type "Setting.Details.SettingList"
23:12:19.700 T:1945243648  NOTICE: UpdateLibraries: Starting video library startup scan
23:12:19.729 T:1945243648  NOTICE: initialize done
23:12:19.729 T:1945243648  NOTICE: Running the application...
23:12:19.773 T:1945243648  NOTICE: starting zeroconf publishing
23:12:19.774 T:1945243648  NOTICE: CWebServer[8080]: Started
23:12:19.774 T:1945243648  NOTICE: starting upnp client
23:12:19.782 T:1645212576  NOTICE: ES: Starting UDP Event server on port 9777
23:12:19.783 T:1645212576  NOTICE: UDP: Listening on port 9777 (ipv6 : true)
23:12:20.200 T:1735304096  NOTICE: VideoInfoScanner: Starting scan ..
23:12:20.221 T:1735304096  NOTICE: VideoInfoScanner: Finished scan. Scanning for video info took 00:00
23:12:21.026 T:1760469920   ERROR: GetDirectory - Error getting /storage/.kodi/temp/plugin.video.youtube/
23:12:21.750 T:1725780896 WARNING: CSkinInfo: failed to load skin settings
23:12:21.974 T:1752081312   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <class 'socket.error'>
                                            Error Contents: (19, 'No such device')
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/script.tubecast/main.py", line 7, in <module>
                                                service.run()
                                              File "/storage/.kodi/addons/script.tubecast/resources/lib/service.py", line 20, in run
                                                ssdp_server.start(interfaces=Kodicast.interfaces)
                                              File "/storage/.kodi/addons/script.tubecast/resources/lib/tubecast/ssdp.py", line 153, in start
                                                self.server = MulticastServer((self.SSDP_ADDR, self.SSDP_PORT), SSDPHandler, interfaces=interfaces)
                                              File "/storage/.kodi/addons/script.tubecast/resources/lib/tubecast/ssdp.py", line 57, in __init__
                                                self.handle_membership(socket.IP_ADD_MEMBERSHIP)
                                              File "/storage/.kodi/addons/script.tubecast/resources/lib/tubecast/ssdp.py", line 85, in handle_membership
                                                cmd, mreq)
                                              File "/usr/lib/python2.7/socket.py", line 228, in meth
                                            error: (19, 'No such device')
                                            -->End of Python script error report<--
23:12:22.459 T:1725780896   ERROR: ## LibreELEC Addon ## oe::load_url(http://releases.libreelec.tv/releases.json) ## ERROR: (URLError(gaierror(-3, 'Temporary failure in name resolution'),))
23:12:22.460 T:1725780896   ERROR: Traceback (most recent call last):
                                              File "/home/chewitt/LibreELEC.82-images/build.LibreELEC-RPi2.arm-8.2.5/LibreELEC-settings-0768930/.install_pkg/usr/share/kodi/addons/service.libreelec.settings/oe.py", line 271, in load_url
                                              File "/usr/lib/python2.7/urllib2.py", line 154, in urlopen
                                              File "/usr/lib/python2.7/urllib2.py", line 429, in open
                                              File "/usr/lib/python2.7/urllib2.py", line 447, in _open
                                              File "/usr/lib/python2.7/urllib2.py", line 407, in _call_chain
                                              File "/usr/lib/python2.7/urllib2.py", line 1228, in http_open
                                              File "/usr/lib/python2.7/urllib2.py", line 1198, in do_open
                                            URLError: <urlopen error [Errno -3] Temporary failure in name resolution>
23:12:22.573 T:1616901024   ERROR: ## LibreELEC Addon ## oe::load_url(https://update.libreelec.tv/updates.php?i=4d3bd6a2ece314e888b618a81d7d247f&d=LibreELEC&pa=RPi2.arm&v=8.2.5&l=0
                                            ) ## ERROR: (URLError(gaierror(-3, 'Temporary failure in name resolution'),))
23:12:22.573 T:1616901024   ERROR: Traceback (most recent call last):
                                              File "/home/chewitt/LibreELEC.82-images/build.LibreELEC-RPi2.arm-8.2.5/LibreELEC-settings-0768930/.install_pkg/usr/share/kodi/addons/service.libreelec.settings/oe.py", line 271, in load_url
                                              File "/usr/lib/python2.7/urllib2.py", line 154, in urlopen
                                              File "/usr/lib/python2.7/urllib2.py", line 429, in open
                                              File "/usr/lib/python2.7/urllib2.py", line 447, in _open
                                              File "/usr/lib/python2.7/urllib2.py", line 407, in _call_chain
                                              File "/usr/lib/python2.7/urllib2.py", line 1241, in https_open
                                              File "/usr/lib/python2.7/urllib2.py", line 1198, in do_open
                                            URLError: <urlopen error [Errno -3] Temporary failure in name resolution>
23:12:22.580 T:1616901024   ERROR: ## LibreELEC Addon ## system::check_updates_v2 ## ERROR: (TypeError('expected string or buffer',))
23:12:22.581 T:1616901024   ERROR: Traceback (most recent call last):
                                              File "/home/chewitt/LibreELEC.82-images/build.LibreELEC-RPi2.arm-8.2.5/LibreELEC-settings-0768930/.install_pkg/usr/share/kodi/addons/service.libreelec.settings/resources/lib/modules/system.py", line 768, in check_updates_v2
                                              File "/usr/lib/python2.7/json/__init__.py", line 339, in loads
                                              File "/usr/lib/python2.7/json/decoder.py", line 364, in decode
                                            TypeError: expected string or buffer
09:46:24.944 T:1743692704  NOTICE: WakeOnAccess [paisley.local] trigged by accessing : smb://paisley.local/bookshelf/Music/陶喆 - 陶喆 (1997)/01 Airport Take Off.mp3
09:46:25.082 T:1735304096  NOTICE: WakeOnAccess [paisley.local] trigged by accessing : smb://paisley.local/bookshelf/Music/張懸/2006 - My Life Will.._/01 - Scream.mp3
09:46:25.098 T:1584472992  NOTICE: WakeOnAccess [paisley.local] trigged by accessing : smb://paisley.local/bookshelf/Music/Zero 7/2001 - Simple Things/03 - Destiny.mp3
09:46:25.153 T:1576084384  NOTICE: WakeOnAccess [paisley.local] trigged by accessing : smb://paisley.local/bookshelf/Music/Zeigeist/2008 - The Jade Motel/07 - Humanitarianism.mp3
09:46:25.167 T:1735304096  NOTICE: WakeOnAccess success exit, server already running
09:46:45.694 T:1735304096  NOTICE: Previous line repeats 3 times.
09:46:45.695 T:1735304096   ERROR: GetDirectory - Error getting
09:47:49.653 T:1645212576   ERROR: Previous line repeats 3 times.
09:47:49.653 T:1645212576  NOTICE: ES: Client  from ::ffff:192.168.1.109 timed out
10:00:27.543 T:1519383456   ERROR: GetDirectory - Error getting
10:02:01.297 T:1945243648  NOTICE: Samba is idle. Closing the remaining connections
21:59:31.355 T:1714332576  NOTICE: WakeOnAccess [paisley.local] trigged by accessing : smb://paisley.local/bookshelf/Music/Barefoot/2006 - Barefoot/01 - Don't Do It.mp3
21:59:31.383 T:1546560416  NOTICE: WakeOnAccess [paisley.local] trigged by accessing : smb://paisley.local/bookshelf/Music/Dengue Fever - The Deepest Lake (2015)/01 - Tokay.mp3
21:59:31.404 T:1714332576  NOTICE: WakeOnAccess success exit, server already running
21:59:54.103 T:1546560416  NOTICE: Previous line repeats 1 times.
21:59:54.103 T:1546560416   ERROR: GetDirectory - Error getting
22:01:23.818 T:1735304096   ERROR: Previous line repeats 6 times.
22:01:23.818 T:1735304096  NOTICE: SaveMACDiscoveryResult - Mac discovered for host 'paisley.local' -> 'B8:27:EB:68:F3:B4'
22:01:24.021 T:1945243648  NOTICE: QueueMACDiscoveryForHost - skip Mac discovery for non-local host 'k.mjh.nz'
22:01:24.084 T:1714332576  NOTICE: VideoInfoScanner: Starting scan ..
22:01:26.002 T:1714332576   ERROR: CCurlFile::Open failed with code 404 for https://api.thetvdb.com/search/series?name=Community%20S01E06%20Football%20Feminism%20And%20You|Authorization=Bearer%20eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE1MzA1NDAwODUsImlkIjoiS29kaSIsIm9yaWdfaWF0IjoxNTMwNDUzNjg1fQ.Uj1TZohkkM3AixiEQfO7l5fTAADrQw-iVZJLfx2EVSBzGNoStO6iC0BNOa5uYo-BF3IlyeDaFYb_56ar39xlQu75grN-ghHCJQhRcORWVxUNdPUGToLGn0vmLopk-f-0UsWZr_zOPzHvf2nD4pssNZBpvQXmUxDbXaLtCpDjMTegQQUMZ9XYltTsbsgY5ws1S8l1lBqEYXIyISiqWFYrH_hopKPT5t8BAIuY-omaYISlMoEN-JOeY1U276t5zRIYN3uvM_4zhg5uJOAwI_oGFp6fEvp0oQACf15wmaH-BLZocrfQpRUDgkcTmoU5w9s5wxVX5itFd-g3_EKevcfDEg&accept-language=en
22:01:26.002 T:1714332576   ERROR: Run: Unable to parse web site
22:01:26.848 T:1714332576   ERROR: CCurlFile::Open failed with code 404 for https://api.thetvdb.com/search/series?name=Community.S01E06.Football.Feminism.And.You|Authorization=Bearer%20eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE1MzA1NDAwODUsImlkIjoiS29kaSIsIm9yaWdfaWF0IjoxNTMwNDUzNjg1fQ.Uj1TZohkkM3AixiEQfO7l5fTAADrQw-iVZJLfx2EVSBzGNoStO6iC0BNOa5uYo-BF3IlyeDaFYb_56ar39xlQu75grN-ghHCJQhRcORWVxUNdPUGToLGn0vmLopk-f-0UsWZr_zOPzHvf2nD4pssNZBpvQXmUxDbXaLtCpDjMTegQQUMZ9XYltTsbsgY5ws1S8l1lBqEYXIyISiqWFYrH_hopKPT5t8BAIuY-omaYISlMoEN-JOeY1U276t5zRIYN3uvM_4zhg5uJOAwI_oGFp6fEvp0oQACf15wmaH-BLZocrfQpRUDgkcTmoU5w9s5wxVX5itFd-g3_EKevcfDEg&accept-language=en
...
```

Let me know if there's any other way I can get useful log information to you.

**Comment by siku2** [comment]
I just started investigating #13 and I noticed that when TubeCast is run without internet connection it produces this exact issue. Combined with the fact that the log is followed by other network issues such as "Temporary failure in name resolution" which typically point to a lack of internet connectivity and also that they were unable to reproduce the issue since it first occurred I would suggest that it was nothing but that - a lack of internet connection.

The other option would be that [multicast support isn't supported by the network interface](https://stackoverflow.com/a/3188060) but I find that unlikely as I was unable to find anyone else having issues with multicast in LibreElec or on a RPi 3b+. 
I suggest this issue be closed in favour of #13.
